docs/reference/superjson.mdx

```diff
@@ -0,0 +1,294 @@
+---
+slug: /reference/superjson
+---
+
+# Super.json Reference
+
+Super.json is the main Superface configuration file. It contains the configuration about profiles, maps and providers used.
+
+## Anatomy
+
+At the top level, super.json file consists of two sections: [profiles](#profiles) and [providers](#providers). the main purpose profile section is to set up profiles which are to be used, their versions and/or file locations, and their default values. The purpose of provider section is to set up providers that are to be used and their credentials.
+
+```json
+{
+  "profiles": {
+    // ...
+  },
+  "providers": {
+   // ...
+  }
+}
+```
+
+## Profiles
+
+The profiles section consists of a record of all installed profiles. The key is the name of the profile and value is either semantic version string, a file URI or a [ProfileSettings](#profilesettings) object.
+
+```json
+{
+  "profiles": {
+    "myprofile1": "1.0.0",
+    "myprofile2": "file:path/to/profile.supr",
+    "myprofile3": {
+      // ...
+    }
+  }
+}
+```
+
+### ProfileSettings
+
+The ProfileSettings object contains either `file` or `version` property, analogous to the shortcut strings above and following optional properties:
+
+- `priority`: an array of strings containing provider names in desired order for failover
+- `providers`: an array of [ProfileProviderSettings](#profileprovidersettings) objects or URI strings, used for configuring providers for the profile. Not to be confused with top-level Providers record
+- `defaults`: a record of defaults for the usecase, key is usecase name, value is [UsecaseDefaults](#usecasedefaults) object
+
+```json
+{
+  "version": "1.0.0",
+  "priority": ["provider1", "provider2"],
+  "defaults": {
+    "myusecase": {
+      // ...
+    }
+  },
+  "providers": {
+    "provider1": {
+      // ...
+    },
+    "provider2": "file:path/to/provider.json"
+  }
+}
+```
+
+### UsecaseDefaults
+
+UsecaseDefaults is a record used to configure default values for a usecase, independent of provider. The key is usecase name and the value is an object with following optional properties.
+
+- `input`: an object providing default values according to usecase's input
+- `providerFailover`: a boolean specific whether provider failover should be allowed for the usecase
+
+```json
+{
+  "myusecase": {
+    "input": {
+      "someInputProperty": 42
+    },
+    "providerFailover": true
+  }
+}
+```
+
+### ProfileProviderSettings
+
+ProfileProviderSettings object is used for configuring specific providers for a profile. It can point to either file or specific variant and revision. It contains following optional property:
+
+- `defaults`: a record of defaults per usecase, key is usecase name and value is a [ProfileProviderDefaults](#profileproviderdefaults) object
+
+```json
```

Comment:
Provide explaining titles for these examples. Also it's hard to navigate how deeply nested we are in the super.json file, so preferably provide the whole context and use lines highlighting to pinpoint the relevant part.